lib/IRGen/GenMeta.cpp

```diff
@@ -1075,6 +1075,9 @@ void irgen::emitLazyCacheAccessFunction(IRGenModule &IGM,
   // FIXME: Technically should be "consume", but that introduces barriers in the
   // current LLVM ARM backend.
   auto load = IGF.Builder.CreateLoad(cache);
+  if (IGM.Opts.Sanitize == SanitizerKind::Thread)
+    load->setOrdering(llvm::AtomicOrdering::Acquire);
+
```

Comment:
Please leave a comment like:

  // Make this barrier explicit when building for TSan to avoid false positives.